comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -121,11 +121,16 @@ func (c *collector) Start(ctx context.Context, store workloadmeta.Component) err
 }
 
 func (c *collector) setTaskCollectionParser() {
-	_, err := ecsmeta.V4FromCurrentTask()
-	if c.taskCollectionEnabled && err == nil {
+	ok, err := ecsmeta.IsMetadataV4Available()
+	if c.taskCollectionEnabled && ok && err == nil {
+		log.Infof("detailed task collection enabled, using metadata v4 endpoint")
 		c.taskCollectionParser = c.parseTasksFromV4Endpoint
 		return
+	} else if c.taskCollectionEnabled && err != nil {
+		log.Warnf("detailed task collection enabled but agent cannot determine if v4 metadata endpoint is available: %s", err.Error())
 	}
+
+	log.Infof("detailed task collection disabled, using metadata v1 endpoint")
 	c.taskCollectionParser = c.parseTasksFromV1Endpoint
 }
```

Comment:
IMHO I think the code can be made simpler, and some function calls can be avoided. 

For example, if c.taskCollectionEnabled is `false`, we don't really need to check if metadata v4 is available. 
If we do a simple check on c.taskCollectionEnabled at first, it can simplify the logic and reduce the nesting. 

I mean something like the following:

```
if !c.taskCollectionEnabled {
      c.taskCollectionParser = c.parseTasksFromV1Endpoint
      return
}

ok, err := escmeta.IsMetadataV4Available()

if err != nil || !ok {
c.taskCollectionParser = c.parseTasksFromV1Endpoint
} else {
c.taskCollectionParser = c.parseTasksFromV4Endpoint
}
```

WDYT?